mesos-client/src/main/scala/com/mesosphere/mesos/client/Session.scala

```diff
@@ -40,7 +44,15 @@ case class Session(baseUri: Uri, streamId: String, authorization: Option[Credent
       .watch(ref)
       .mapAsync(parallelism) {
         case (el, ctx) =>
-          akka.pattern.ask(ref).?(el)(timeout).mapTo[HttpResponse](tag).map(o => (o, ctx))
+          akka.pattern
+            .ask(ref)
+            .?(el)(timeout)
+            .transform {
+              case Failure(ex: AskTimeoutException) => Failure(SessionRequestException("Mesos request timed out.", ex))
```

Comment:
This is just illustrating an issue. The `Http().singleRequest` might not complete in time an trigger this `AskTimeoutException`. It also leaves a dead letter. I do not like it. @timcharper, how should we do this? I'd say we increase the ask time out and let Akka trigger a client timeout. I just don't know which of [these](https://doc.akka.io/docs/akka-http/current/common/timeouts.html) we have to watch for.

Reply:
The session actor should reply with a `Future` immediately.

Reply:
And convert it to a typed actor.